src/ansys/aedt/core/modeler/cad/primitives.py

```diff
@@ -8438,23 +8438,28 @@ def _check_material(self, matname, defaultmatname, threshold=100000):
 
         # Note: Material.is_dielectric() does not work if the conductivity
         # value is an expression.
-        if isinstance(matname, Material):
+        if isinstance(material, Material):
             if self._app._design_type == "HFSS":
-                return matname.name, matname.is_dielectric(threshold)
+                return material.name, material.is_dielectric(threshold)
             else:
-                return matname.name, True
-        if matname:
-            if self._app.materials[matname]:
+                return material.name, True
+        if material:
+            if "[" in material or "(" in material:
+                array = material.split("[") or material.split("(")
+                if array[0] in self._app.variable_manager.design_variables.keys():
+                    index = int(array[1].strip("]"))
+                    material = self._app.variable_manager.design_variables[array[0]].numeric_value[index]
```

Comment:
Is there a reason why ")" isn't handled ? Feels like it is missing as it was handled in previous lines of code.